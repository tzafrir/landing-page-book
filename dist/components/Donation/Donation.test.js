'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Donation = require('./Donation');

var _Donation2 = _interopRequireDefault(_Donation);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Donation', function () {
  var props = void 0;
  var mounted = void 0;
  var donation = function donation() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Donation2.default, props));
    }
    return mounted;
  };
  describe('Donation', function () {
    beforeEach(function () {
      props = {
        url: 'https://front10.com'
      };
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Donation2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = donation().find('img');

      expect(divs.length).toBe(0);
    });
  });
});