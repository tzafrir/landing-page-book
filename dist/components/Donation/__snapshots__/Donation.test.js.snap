// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donation Donation should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Donation
    image="https://opencollective.com/webpack/donate/button@2x.png?color=blue"
    imageAlt="Open collective logo"
    imageWidth="300"
    url="https://front10.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Image
        alt="Open collective logo"
        border={false}
        className="img-fluid"
        imgFilter={null}
        rounded={false}
        shadow={false}
        src="https://opencollective.com/webpack/donate/button@2x.png?color=blue"
        tooltip=""
        width="300"
      />,
      "className": "",
      "disabled": false,
      "href": "https://front10.com",
      "style": null,
      "target": "_blank",
      "tooltip": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alt": "Open collective logo",
        "border": false,
        "className": "img-fluid",
        "imgFilter": null,
        "rounded": false,
        "shadow": false,
        "src": "https://opencollective.com/webpack/donate/button@2x.png?color=blue",
        "tooltip": "",
        "width": "300",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Image
          alt="Open collective logo"
          border={false}
          className="img-fluid"
          imgFilter={null}
          rounded={false}
          shadow={false}
          src="https://opencollective.com/webpack/donate/button@2x.png?color=blue"
          tooltip=""
          width="300"
        />,
        "className": "",
        "disabled": false,
        "href": "https://front10.com",
        "style": null,
        "target": "_blank",
        "tooltip": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Open collective logo",
          "border": false,
          "className": "img-fluid",
          "imgFilter": null,
          "rounded": false,
          "shadow": false,
          "src": "https://opencollective.com/webpack/donate/button@2x.png?color=blue",
          "tooltip": "",
          "width": "300",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
